politeiad/backendv2/tstorebe/store/localdb/localdb.go

```diff
@@ -0,0 +1,171 @@
+// Copyright (c) 2020-2021 The Decred developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+package localdb
+
+import (
+	"errors"
+	"fmt"
+	"path/filepath"
+	"sync"
+
+	"github.com/decred/politeia/politeiad/backendv2/tstorebe/store"
+	"github.com/decred/politeia/util"
+	"github.com/syndtr/goleveldb/leveldb"
+)
+
+const (
+	// encryptionKeyFilename is the filename of the encryption key that
+	// is created in the store data directory.
+	encryptionKeyFilename = "leveldb-sbox.key"
+)
+
+var (
+	_ store.BlobKV = (*localdb)(nil)
+)
+
+// localdb implements the store BlobKV interface using leveldb.
+//
+// NOTE: this implementation was created for testing. The encryption techniques
+// used may not be suitable for a production environment. A random secretbox
+// encryption key is created on startup and saved to the politeiad application
+// dir. Blobs are encrypted using random 24 byte nonces.
+type localdb struct {
+	sync.RWMutex
```

Comment:
This is only needed for access; the key should not need protection. We do not want to encrypt with a mutex held.